packages/core/bootstrap/src/lib/ws/epics.ts

```diff
@@ -76,6 +77,62 @@ type ConnectRequestedActionWithState = [
   },
 ]
 
+export const writeMessageToCacheEpic: Epic<AnyAction, AnyAction, { ws: RootState }, any> = (
```

Comment:
Nice! We need even more of this. (not for this PR, but great start)

Could we move this so that the order of reading this file is closer to their execution order?
(e.g. put this below `connectEpic`)